frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts

```diff
@@ -0,0 +1,271 @@
+      },
+    });
+    return () => {
+      // Unsubscribe when the component is unmounted.
+      void observer.unsubscribe();
+    };
+  }, []);
+
+  useEffect(() => {
+    const { pathname } = currentLocation;
+    const parts = pathname.split('/');
+    const pageId = parts[parts.length - 1];
+    setActivePageId(pageId);
+  }, [currentLocation]);
+
+  useEffect(() => {
+    if (page.showPagesInside) {
+      setFolderHeight(`${PAGE_ITEM_HEIGHT + getChildCount(page) * PAGE_ITEM_HEIGHT}px`);
+    } else {
+      setFolderHeight(`${PAGE_ITEM_HEIGHT}px`);
+    }
+  }, [page, pages]);
+
+  // recursively get all unfolded child pages
+  const getChildCount: (startPage: IPage) => number = (startPage: IPage) => {
+    let count = 0;
+    count = pages.filter((p) => p.parentPageId === startPage.id).length;
+    pages
+      .filter((p) => p.parentPageId === startPage.id)
+      .forEach((p) => {
+        if (p.showPagesInside) {
+          count += getChildCount(p);
+        }
+      });
+    return count;
+  };
+
+  const onUnfoldClick = () => {
+    appDispatch(pagesActions.toggleShowPages({ id: page.id }));
+  };
+
+  const onPageOptionsClick = () => {
+    setShowPageOptions(!showPageOptions);
+  };
+
+  const startPageRename = () => {
+    setShowRenamePopup(true);
+    closePopup();
+  };
+
+  const onNewPageClick = () => {
+    setShowNewPageOptions(!showNewPageOptions);
+  };
+
+  const changePageTitle = async (newTitle: string) => {
+    await service.update({ name: newTitle });
+    appDispatch(pagesActions.renamePage({ id: page.id, newTitle }));
+  };
+
+  const closeRenamePopup = () => {
+    setShowRenamePopup(false);
+  };
+
+  const deletePage = async () => {
+    closePopup();
+    await service.delete();
+    appDispatch(pagesActions.deletePage({ id: page.id }));
+  };
+
+  const duplicatePage = async () => {
+    closePopup();
+    await service.duplicate();
+  };
+
+  const closePopup = () => {
+    setShowPageOptions(false);
+    setShowNewPageOptions(false);
+  };
+
+  const onPageClick = (page: IPage) => {
+    const pageTypeRoute = (() => {
+      switch (page.pageType) {
+        case ViewLayoutPB.Document:
+          return 'document';
+        case ViewLayoutPB.Grid:
+          return 'grid';
+        case ViewLayoutPB.Board:
+          return 'board';
+        default:
+          return 'document';
+      }
+    })();
+
+    navigate(`/page/${pageTypeRoute}/${page.id}`);
+  };
+
+  const onAddNewDocumentPage = async () => {
+    closePopup();
+    if (!workspace?.id) return;
+    const workspaceService = new WorkspaceBackendService(workspace.id);
+    const newViewResult = await workspaceService.createView({
+      name: 'New Document 1',
+      layoutType: ViewLayoutPB.Document,
+      parentViewId: page.id,
+    });
+    if (newViewResult.ok) {
+      try {
+        const newView = newViewResult.val;
+        const c = new DocumentController(newView.id);
+        await c.create();
+        await c.dispose();
```

Comment:
This code needs to be deleted, it has been deleted on the main branch

Reply:
done